cron-ci/build-hpc-stack.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -l
+# pass -l (treat script as login shell) so lmod module environment is picked up
+set -eux
+
+# Parse hpc-stack yaml file to get version variables
+source stack_helpers.sh
+eval $(parse_yaml ${HPC_STACK_FILE} "STACK_")
+
+# get the current hpc-stack hash and compare it to the previously built hash (if it exists)
+# if it exists and matches current hash, don't build
+current_hash=$(git rev-parse HEAD)
+
+build=true
+if [[ -f "${HPC_HOMEDIR}/prev_hash.txt" ]]; then
+    prev_hash=$(cat ${HPC_HOMEDIR}/prev_hash.txt)
+    if [[ "$current_hash" == "$prev_hash" ]]; then
+        build=false
+        echo `date`
+        echo ""
+        echo "hpc-stack version has not changed since last time. Not building."
+        echo "hpc-stack hash: ${current_hash}"
+    fi
+fi
```

Comment:
Can you move this section to `setup-cron.sh`

Reply:
Yep

Reply:
and then, if the hash is updated, call this script and the rest after than.
If it is not, exit there and then